azure-iot-device/tests/provisioning/pipeline/test_provisioning_pipeline.py

```diff
@@ -327,226 +312,129 @@ def test_op_fail(self, mocker, pipeline, arbitrary_exception):
+        op = pipeline._pipeline.run_op.call_args[0][0]
 
-        fake_publish_topic = "$dps/registrations/PUT/iotdps-register/?$rid={}".format(
-            fake_request_id_1
-        )
-        mock_mqtt_transport.wait_for_publish_to_be_called()
-        assert mock_mqtt_transport.publish.call_count == 1
-        assert mock_mqtt_transport.publish.call_args[1]["topic"] == fake_publish_topic
-        assert mock_mqtt_transport.publish.call_args[1]["payload"] == fake_register_publish_payload
-
-        # while we're waiting for that send to complete, send another event
-        callback_2 = mocker.MagicMock()
-        # provisioning_pipeline.send_message(fake_msg_2, callback_2)
-        mock_provisioning_pipeline.register(payload=fake_msg_2, callback=callback_2)
-
-        # verify that we've called publish twice and verify that neither send_message
-        # has completed (because we didn't do anything here to complete it).
-        mock_mqtt_transport.wait_for_publish_to_be_called()
-        assert mock_mqtt_transport.publish.call_count == 2
-        callback_1.assert_not_called()
-        callback_2.assert_not_called()
-
-    @pytest.mark.it("Connects , sends request queues and then disconnects")
-    def test_connect_send_disconnect(self, mock_provisioning_pipeline, mock_mqtt_transport):
-        # connect
-        mock_provisioning_pipeline.connect()
-        mock_mqtt_transport.on_mqtt_connected_handler()
-        mock_provisioning_pipeline.wait_for_on_connected_to_be_called()
-
-        # send an event
-        mock_provisioning_pipeline.register(payload=fake_mqtt_payload)
-        mock_mqtt_transport.on_mqtt_published(0)
-
-        # disconnect
-        mock_provisioning_pipeline.disconnect()
-        mock_mqtt_transport.disconnect.assert_called_once_with()
+        op.complete(error=arbitrary_exception)
+        assert cb.call_count == 1
+        assert cb.call_args == mocker.call(error=arbitrary_exception)
 
 
-@pytest.mark.describe("ProvisioningPipeline - .disconnect()")
-class TestDisconnect(object):
-    @pytest.mark.it("Calls disconnect on provider")
-    def test_disconnect_calls_disconnect_on_provider(
-        self, mock_provisioning_pipeline, mock_mqtt_transport
-    ):
-        mock_provisioning_pipeline.connect()
-        mock_mqtt_transport.on_mqtt_connected_handler()
-        mock_provisioning_pipeline.wait_for_on_connected_to_be_called()
-        mock_provisioning_pipeline.disconnect()
+@pytest.mark.describe("ProvisioningPipeline - .enable_responses()")
+class TestEnable(object):
+    @pytest.mark.it("Marks the feature as enabled")
+    def test_mark_feature_enabled(self, pipeline, mocker):
+        assert not pipeline.responses_enabled[feature]
+        pipeline.enable_responses(callback=mocker.MagicMock())
+        assert pipeline.responses_enabled[feature]
 
-        mock_mqtt_transport.disconnect.assert_called_once_with()
+    @pytest.mark.it("Runs a EnableFeatureOperation on the pipeline")
```

Comment:
>EnableFeatureOperation [](start = 28, length = 22)

We should also validate the feature name on the operation, as we do in the hub's mqtt pipeline.

Except, as pointed out in another comment, for some reason it doesn't have one here? We probably shouldn't be enabling a 'None' feature, and just always taking that to be the single item we can enable.